File: CS50x/1. Apna College - CS50x/12.c
/*

> Date Created: June 15, 2023
> Author: Ishaan Rastogi
> Purpose: This code is to showcase relational operators' working
> Operating System: This is only for Windows OS, it may or may not work on other OS
> Program Status: 100% Working

*/

/*
    
    Topic: Control Instructions - Used to determine flow of program
    
    a> Sequence Control- Runs sequentially ( All the programmes till now contained this CIs )
    
    b> Decision Control- Topics like If else
    
    c> Loop Control- Topics like For, while loops
    
    d> Case Control- Depends on different cases

    Topic: Types of Operators

    a> Arithmatic- +, -, *, /, %
    
    b> Relational- ==, >, >=, <, <=, !=
            != denotes not equal while == denotes equal
    
    c> Logical- They just work like logic gates work.
                    && (AND), || (OR), ! (NOT)
    
    >>> Operator Precedence Order:
      
                    (NOT) > (* , / , %) > ( + , - ) > (<, <=, >, >=) > ( EQUAL TO, NOT EQUAL TO) > (AND) > (OR) > (=) (Assignment operator)

    d> Assignment- = , Short hand operators { +=, -=, *=, /=, %= }
    
    e> Bitwise- ^, &

    f> Ternary- Conditional operators
    */
   

#include<stdio.h>
int main() {
    
// Binary Output: 1 denotes True & 0 denotes False values

    printf("%d \n", 4 == 4);
    printf("%d \n\n", 4 == 3);
    
    printf("%d \n", 3 > 4 && 5 > 2 ); // AND (&&)
    printf("%d \n\n", 3 > 4 || 5 > 2 ); // OR (||)
    
    printf("%d \n", !( 5 > 3) );
    printf("%d \n\n", !( 5 > 3) && ( 3 > 4 ) ); // Precedence- NOA Not And Or

    int a = 1;
    int b = 2;
    a += b; // Instead of this a = a + b;
    printf("%d \n", a);

}

/* 

Ctrl + Shift + `

Terminal Code Loading...

gcc 12.c
./a.exe

 */

File: Web Development Projects/15 Intermediate JS/12.js
/*
> Date Created: June 10, 2024
> Author: Ishaan Rastogi
> Purpose: Better version of 10.js using for loop
> Operating System: This is only for Windows OS, it may or may not work on other OS
> Program Status: 100% Working
*/

/*

While is used when you need to state & for when you need to iterate the function upto a certain number.

*/

var output = [];

//Creating function to push new numbers
function fizzBuzz() {
    for (var c = 1; c < 101; c++) {
        if ( c % 3 === 0) {
            output.push("Fizz");
        }
        else if ( c % 5 === 0) {
            output.push("Buzz");
        }
        else if ( c % 3 === 0 && c % 5 === 0) {
            output.push("FizzBuzz");
        }
        else {
            output.push(c);
        }
    }
    console.log(output);
}

/* 
For Terminal Code... Ctrl + Shift + `
*/